boxsdk/session/box_session.py

```diff
@@ -248,6 +248,8 @@ def _raise_on_unsuccessful_request(self, network_response, expect_json_response,
                 url=url,
                 method=method,
                 context_info=response_json.get('context_info', None),
+                request_response=network_response.request_response,
```

Comment:
This is an unnecessary backwards incompatibility, since custom networks might not have `request_response`.

Instead, I would propose that this line be:

`network_response=network_response`

This has the benefits of:
- Is backwards compatible.
- Works with custom subclasses of `NetworkResponse`.
- Still allows the user to access `network_response.request_response`, if `network_response` is an instance of `DefaultNetworkResponse`.
- Not tying us to the `requests` library even further.
- Probably never again needing to add more fields to the exception, since we now include a maximal amount of information.

We could also avoid adding `access_token_used`, since the user could use `exception.network_response.access_token_used` instead. It's a minor difference, we could do either. Having it directly on the object is slightly more clear and direct and less work for the user to access, but is another property that we would have to maintain forever for backwards compatibility.